### ISSUE TITLE
Update rustfmt configuration

### ISSUE THREAD

**Issue by ishitatsuyuki** [RustAudio/cpal#302]
The `.rustfmt.toml` file is now containing a lot of obsolete options, and one even causes rustfmt to fail.

```
Error: Decoding config file failed:
unknown variant `Visual`, expected one of `Compressed`, `Tall`, `Vertical` for key `fn_args_layout`
Please check your config file.
```

Updating the file to adopt changes in rustfmt would be appreciated.

**Comment by mitchmindtree** [comment]
Agreed - my personal preference is normally just to go without a `.rustfmt.toml` altogether and use the defaults. It also might be worth adding a formatting check to the CI too as a step towards maintaining formatting consistency throughout the crate. Would you be happy with this? Maybe we can address this after your current stream overhaul to avoid causing any major conflicts.

**Comment by ishitatsuyuki** [comment]
I checked a few projects that enforces style in CI - clippy and rust-analyzer - and they both had a rustfmt config in repository. I guess it’s worth the time to preserve the current configuration as much as possible.

And yeah, let’s add the check in CI after the implementing period.

**Comment by mitchmindtree** [comment]
This has since been addressed by #365
